pipeline/load.py

```diff
@@ -0,0 +1,314 @@
+"""Script containing functions for loading sales data into the Database."""
+
+from os import environ as ENV
+import logging
+from typing import Any, Dict, List, Optional
+import psycopg2
+import psycopg2.extras
+from psycopg2.extensions import connection as DBConnection, cursor as DBCursor
+from dotenv import load_dotenv
+
+from extract import get_sales_data
+from transform import transform_sales_data
+
+REQUIRED_FIELDS_TRACK = ["album_url", "track_tags"]
+REQUIRED_FIELDS_SINGLE = ["track_tags"]
+REQUIRED_FIELDS_ALBUM = ["album_tags"]
+
+
+def get_connection() -> DBConnection:
+    """Creates a database session and returns a connection object."""
+    return psycopg2.connect(
+        host=ENV["DB_HOST"],
+        port=ENV["DB_PORT"],
+        user=ENV["DB_USER"],
+        password=ENV["DB_PASS"],
+        database=ENV["DB_NAME"],
+    )
+
+
+def get_cursor(connection: DBConnection) -> DBCursor:
+    """Creates and returns a cursor to execute PostgreSQL commands."""
+    return connection.cursor(cursor_factory=psycopg2.extras.DictCursor)
+
+
+def get_or_insert_artist(cursor: DBCursor, artist_name: str, artist_url: str) -> int:
+    """Adds the artist to the artist table and returns the artist_id if not present,
+    otherwise gets the artist_id from the table"""
+
+    cursor.execute(
+        "SELECT artist_id FROM artist WHERE artist.name = %s AND artist.url = %s",
+        (
+            artist_name,
+            artist_url,
+        ),
+    )
+    artist_id = cursor.fetchone()
+
+    if artist_id:
```

Comment:
Should probably have an ifinstance to ensure that what you're getting is a list.

Reply:
I think because the id is coming from the database where it is stored as an int, its okay to assume whats being fetched is in the correct format